nemo/core/classes/modelPT.py

```diff
@@ -606,10 +606,48 @@ def setup_optimizer_param_groups(self):
             See https://pytorch.org/docs/stable/optim.html for more information.
             By default, ModelPT will use self.parameters().
             Override this method to add custom param groups.
-    """
-        param_groups = None
-        if hasattr(self, 'parameters'):
-            param_groups = [{'params': self.parameters()}]
+            In the config file, add 'optim_param_groups' to support different LRs 
+            for different components (unspecified params will use the default LR):
+            model:
+                optim_param_groups:
+                    encoder: 1e-4
+                    decoder: 1e-3
+                optim:
+                    lr: 3e-3   
+        """
+        if not hasattr(self, "parameters"):
+            self._optimizer_param_groups = None
+            return
+
```

Comment:
Btw, just realized, why are we caching the param groups ? Don't do it, it silently causes issues with adapters. Let's return the value instead and update setup optimization to use returned value. 

@ericharper is that fine ? Do we need to make changes in Megatron side ?

Reply:
Yes, let's change it if possible. Caching causes downstream problems with adapters c